packages/cactus-api-client/src/main/typescript/verifier.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2020-2021 Hyperledger Cactus Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Verifier.ts
+ */
+
+// TODO - Document classes and methods
+
+import { Subscription } from "rxjs";
+
+import { Logger } from "@hyperledger/cactus-common";
+import { LogLevelDesc, LoggerProvider } from "@hyperledger/cactus-common";
+import { ISocketApiClient } from "@hyperledger/cactus-core-api";
+
+export interface VerifierEventListener<BlockType> {
+  onEvent(ledgerEvent: BlockType): void;
```

Comment:
@outSH Would it make sense to have an onError handler here so that it can used by the callsite when something bad happens (I saw somewhere else in the code that right now it just logs errors out and then moves along without doing anything else)

Reply:
Yes, I've even included this in the TODO comment below in this file, I wanted to add a required onError callback in the following refactor PR. Now, when I think of it, it's more reasonable to add onError optional callback, which can be added now and will not trigger changes in multiple places through the code :)

Reply:
Added optional onError callback and it's unit test in the most recent version.